src/client/java/net/neoforged/neoforge/client/event/RegisterSpriteDefaultMetadataSectionTypesEvent.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.event;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.Executor;
+import net.minecraft.client.gui.GuiSpriteManager;
+import net.minecraft.client.renderer.texture.SpriteLoader;
+import net.minecraft.client.renderer.texture.TextureManager;
+import net.minecraft.client.resources.TextureAtlasHolder;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.metadata.MetadataSectionType;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.neoforged.bus.api.Event;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.fml.LogicalSide;
+import net.neoforged.fml.event.IModBusEvent;
+import org.jetbrains.annotations.ApiStatus.Internal;
+
+/**
+ * Fired to allow mods to register their own default {@linkplain MetadataSectionType metadata section types} for use during sprite loading.
+ * This event is fired once on startup, before the initial resource reload.
+ *
+ * <p>
+ * It is important to note that this event only affects texture atlases using {@link SpriteLoader#DEFAULT_METADATA_SECTIONS},
+ * which includes most of the vanilla atlases, and those using {@link SpriteLoader#loadAndStitch(ResourceManager, ResourceLocation, int, Executor)} directly.
```

Comment:
This should be more explicit about which atlases this applies to, something along the lines of "which includes all vanilla atlases in `ModelManager.VANILLA_ATLASES`, all modded atlases registered through `RegisterMaterialAtlasesEvent`, all vanilla and modded atlases using `TextureAtlasHolder(TextureManager, ResourceLocation, ResourceLocation)` and modded atlases using `SpriteLoader#loadAndStitch(ResourceManager, ResourceLocation, int, Executor)`".

Reply:
I've rewritten this part of the documentation and explicitly list these too.